src/lib/TransactionDataFormatting.js

```diff
@@ -19,29 +19,29 @@ class TransactionDataFormatting { // eslint-disable-line no-unused-vars
             return I18n.translatePhrase('tx-data-funding-cashlink');
         }
 
-        // For regular transactions allow interpreting the data as utf8
+        let message = '';
+
+        // For transactions to basic accounts allow interpreting the data as utf8
+        // (the check for no flags is technically redundant, because the chain would
+        // reject any transaction to a basic account with flags)
         if (transaction.flags === Nimiq.TransactionFlag.None
-            && transaction.senderType === Nimiq.AccountType.Basic
             && transaction.recipientType === Nimiq.AccountType.Basic
-            && transaction.senderData.length === 0
             && Utf8Tools.isValidUtf8(transaction.data)) {
-            return Utf8Tools.utf8ByteArrayToString(transaction.data);
+            message = Utf8Tools.utf8ByteArrayToString(transaction.data);
         }
 
         const plainTransaction = transaction.toPlain();
         const plainData = plainTransaction.data;
 
-        // Contract withdrawal without any data
-        // If a transaction has no transaction data but a special sender type, try to give it a useful label nonetheless
-        if (transaction.senderData.length === 0
-            && transaction.data.length === 0
-            && transaction.senderType !== Nimiq.AccountType.Basic) {
-            return I18n.translatePhrase('tx-data-contract-withdrawal', {
+        let prefix = '';
+
+        // Contract withdrawal
+        if (transaction.senderType !== Nimiq.AccountType.Basic) {
+            prefix = I18n.translatePhrase('tx-data-contract-withdrawal', {
                 contractType: TransactionDataFormatting._capitalize(plainTransaction.senderType),
             });
         }
 
-        let prefix = '';
         let includeDataType = true;
```

Comment:
I'd also move the definition of `includeDataType` up to where `prefix` is now defined.